src/vscode-azurefunctions.api.d.ts

```diff
@@ -3,12 +3,16 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import { AppSettingsTreeItem } from "vscode-azureappservice";
+
 export interface AzureFunctionsExtensionApi {
     apiVersion: string;
 
     revealTreeItem(resourceId: string): Promise<void>;
 
     createFunction(options: ICreateFunctionOptions): Promise<void>;
+    downloadAppSettings(node?: AppSettingsTreeItem): Promise<void>;
```

Comment:
You can't guarantee this `AppSettingsTreeItem` class will be the same across extensions, so we can't use it. You can pass something like the function app id and go from there

Reply:
An issue I'm running with that approach is that the tree view is not shared across extensions.  Therefore, even if I pass the application tree id itself, it won't be able to find the node.  As far as I know, the platform team doesn't have plans to expose the Function App associated to the SWA to the Functions API endpoint, so I'm not sure there'll be a way to workaround this issue.

Reply:
A workaround to this is to pass the AzExtTreeDataProvider.  I feel like this makes the API pretty rigid, but considering we're dealing with an Azure resource anyway, I think that's okay.  Let me know if you have any thoughts on it, but that's what I'll implement for now.